src/pynwb/__init__.py

```diff
@@ -289,7 +291,8 @@ def read(self, **kwargs):
         # read the file
         if self.external_resources is not None:
             file = super().read(**kwargs)
-            file.set_external_resources(self.external_resources)
+            er_read=ExternalResources.from_flat_tsv(path=self.external_resources)
+            file.link_resources(er_read)
```

Comment:
To simplify this a bit, I would suggest pulling the common element of reading the NWBFile out of the if/else block such that the if-block only needs to deal with the reading and linking of external resources

```
        # read the file
        file = super().read(**kwargs)
        if self.external_resources is not None:     
            er_read=ExternalResources.from_flat_tsv(path=self.external_resources)
            file.link_resources(er_read)
        return file
```